crates/polars-stream/src/utils/in_memory_linearize.rs

```diff
@@ -0,0 +1,124 @@
+use std::cmp::Reverse;
+use std::collections::BinaryHeap;
+
+use polars_core::frame::DataFrame;
+use polars_core::POOL;
+use polars_utils::priority::Priority;
+use polars_utils::sync::SyncPtr;
+
+use crate::morsel::Morsel;
+
+/// Amount of morsels we need to consider spawning a thread during linearization.
+const MORSELS_PER_THREAD: usize = 256;
+
+/// Given a Vec<Morsel> for each pipe, it will output a vec of the contained dataframes.
+/// If the morsels are ordered by their sequence ids within each vec, and no
+/// sequence ID occurs in multiple vecs, the output will follow the same order globally.
```

Comment:
does `sequence ID occurs in multiple vecs` refer to the inner or outer vec?

Reply:
It must be the outer vec, since that's the only way something can occur in multiple vecs.